src/helpers/shareHelper.js

```diff
@@ -45,9 +45,13 @@ export const shareMessage = (data, query) => {
     contacts: data.contacts
   })?.map(({ url }) => `Link: ${url}`);
 
-  const urlSection = urls.length ? urls.join('\n') : '';
+  let urlSection = urls.length ? urls.join('\n') : '';
   const spacer = urlSection ? '\n' : '';
 
+  if (query === QUERY_TYPES.NEWS_ITEM && data?.sourceUrl?.url) {
+    urlSection = `Link: ${data?.sourceUrl?.url}`;
+  }
```

Comment:
I can update the code in this way. But I didn't think to do it this way because it makes the code more complex. But I can add a comment line explaining this complexity.
```
  const urls = mergeWebUrls({
    webUrls: data.webUrls || (data.sourceUrl && [{ ...data.sourceUrl }]) || [],
    contact: data.contact,
    contacts: data.contacts
  })?.map(({ url }) => `Link: ${url}`);
```

Reply:
what about something like this?

```
const webUrls = data.webUrls.concat(data.sourceUrl);

const urls = mergeWebUrls({
  webUrls,
  contact: data.contact,
  contacts: data.contacts
})?.map(({ url }) => `Link: ${url}`);
```

maybe there is check for source url necessary, but you know what i suggest.

Reply:
Unfortunately this doesn't work. Because data.webUrls is not always an array. But I can do it this way:
```
const webUrls = data.webUrls || (data.sourceUrl && [{ ...data.sourceUrl }]) || [];

  const urls = mergeWebUrls({
    webUrls,
    contact: data.contact,
    contacts: data.contacts
  })?.map(({ url }) => `Link: ${url}`);
```

Reply:
that is the same complex stuff you suggested before, right?
why adding web urls OR source url OR empty array? an empty array does not have any urls.
if there are web urls AND source url, both should result in the urls, right?
please find a solution to have web urls and source url combined as `webUrls` passed to the merger.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/018cf1b3147b95d66f53d626b2de705534f30055